corehq/apps/ota/views.py

```diff
@@ -181,9 +181,25 @@ def claim(request, domain):
 @require_POST
 @check_domain_migration
 def claim_all(request, domain):
-    username = request.POST.get("username")     # username may or may not be fully qualified
-    username = format_username(raw_username(username), domain)
-    user_id = username_to_user_id(username)
+    domain_obj = Domain.get_by_name(domain)
+    if not domain_obj:
+        return HttpResponse(_('Invalid project space "{}".').format(domain), status=500)
+
+    if not request.couch_user.is_member_of(domain_obj, allow_mirroring=True):
+        return HttpResponse(_('{} is not a member of {}.').format(request.couch_user.username, domain), status=500)
```

Comment:
nit: use named format params for the sake of translators

Also changed the return code to 403.

```suggestion
        return HttpResponseForbidden(_('{user} is not a member of {domain}.').format(
            use=request.couch_user.username, domain=domain
        ))
```